components/src/core/AppBar/AppBar.tsx

```diff
@@ -161,6 +161,12 @@ const AppBarRender: React.ForwardRefRenderFunction<unknown, AppBarProps> = (prop
 
     // Adjust the height of the app bar when we cross the scroll thresholds
     useEffect(() => {
+        if (variant === 'collapsed') {
+            setHeight(collapsedHeight);
+        } else if (variant === 'expanded') {
+            setHeight(expandedHeight);
+        }
+
```

Comment:
I might be inclined to pull this out into a separate useEffect hook so that it doesn't get called on every scroll:

```
// update height when variant changes
useEffect(() => {
        if (variant === 'collapsed') {
            setHeight(collapsedHeight);
        } else if (variant === 'expanded') {
            setHeight(expandedHeight);
        } else {
            /* set collapsed or expanded height appropriately for snap variant */
        }
}, [variant]);
```